run/run.py

```diff
@@ -0,0 +1,348 @@
+- To upload results, you must be logged in with `gcloud` and authorized
+"""
+
+def main(platform_id, platform, args, config):
+    assert sys.version_info.major == 3, 'This script requires Python 3.'
+
+    print('PLATFORM_ID:', platform_id)
+    print('PLATFORM INFO:', platform)
+
+    if args.path:
+        print('Running tests in path: %s' % args.path)
+    else:
+        print('Running all tests!')
+
+    if args.upload:
+        print('Setting up storage client')
+        from google.cloud import storage
+        storage_client = storage.Client(project='wptdashboard')
+        bucket = storage_client.get_bucket(config['gs_results_bucket'])
+        verify_gsutil_installed(config)
+
+    if args.create_testrun:
+        assert len(config['secret']) == 64, 'Valid secret required to create TestRun'
+
+    if platform['browser_name'] == 'chrome':
+        browser_binary = config['chrome_binary']
+        webdriver_binary = config['chromedriver_binary']
+    elif platform['browser_name'] == 'firefox':
+        browser_binary = config['firefox_binary']
+        webdriver_binary = config['geckodriver_binary']
+    else:
+        raise
+
+    verify_browser_binary_version(platform, browser_binary)
+    verify_os_name(platform)
+    verify_or_set_os_version(platform)
+
+    print('Platform information:')
+    print('Browser version: %s' % platform['browser_version'])
+    print('OS name: %s' % platform['os_name'])
+    print('OS version: %s' % platform['os_version'])
+
+    print('==================================================')
+    print('Setting up WPT checkout')
+
+    wpt_setup_commands = [
+        ['git', 'reset', '--hard', 'HEAD'], # Needed b/c of keep-wpt-running.patch
+        ['git', 'checkout', 'master'],
+        ['git', 'pull'],
+        ['./manifest', '--work'],
+        # Necessary to keep WPT running on long runs. jeffcarp has a PR out
+        # with this patch: https://github.com/w3c/web-platform-tests/pull/5774
+        # however it needs more work.
+        ['git', 'apply', '%s/util/keep-wpt-running.patch' % config['wptd_path']],
+    ]
+    for command in wpt_setup_commands:
+        return_code = subprocess.check_call(command, cwd=config['wpt_path'])
+        assert return_code == 0, 'Got non-0 return code: %d from command %s' % (return_code, command)
+
+    # TODO(#40): modify this to test against the first SHA of the day
+    CURRENT_WPT_SHA = get_current_wpt_sha(config)
+    print('Current WPT SHA: %s' % CURRENT_WPT_SHA)
+
+    SHORT_SHA = CURRENT_WPT_SHA[0:10]
+
+    LOCAL_REPORT_FILEPATH = "%s/wptd-%s-%s-report.log" % (config['build_path'], SHORT_SHA, platform_id)
+    SUMMARY_PATH = '%s/%s-summary.json.gz' % (SHORT_SHA, platform_id)
+    LOCAL_SUMMARY_GZ_FILEPATH = "%s/%s" % (config['build_path'], SUMMARY_PATH)
+    GS_RESULTS_FILEPATH_BASE = "%s/%s/%s" % (config['build_path'], SHORT_SHA, platform_id)
+    GS_HTTP_RESULTS_URL = 'https://storage.googleapis.com/%s/%s' % (config['gs_results_bucket'], SUMMARY_PATH)
```

Comment:
Although these are meant to be constants, they're not technically globals so lowercase would also be OK